core/modules/replica/SqlCreateTableJob.h

```diff
@@ -0,0 +1,175 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_SQLCREATETABLEJOB_H
+#define LSST_QSERV_REPLICA_SQLCREATETABLEJOB_H
+
+// System headers
+#include <functional>
+#include <list>
+#include <set>
+#include <string>
+#include <tuple>
+
+// Qserv headers
+#include "replica/SqlJob.h"
+
+// This header declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * Class SqlCreateTableJob represents a tool which will broadcast the same request
+ * for creating a new table to all worker databases of a setup. Result sets
+ * are collected in the above defined data structure.
+ */
+class SqlCreateTableJob : public SqlJob {
+public:
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<SqlCreateTableJob> Ptr;
+
+    /// The function type for notifications on the completion of the request
+    typedef std::function<void(Ptr)> CallbackType;
+
+    /// @return the unique name distinguishing this class from other types of jobs
+    static std::string typeName();
+
+    /**
+     * Static factory method is needed to prevent issue with the lifespan
+     * and memory management of instances created otherwise (as values or via
+     * low-level pointers).
+     *
+     * @param database
+     *   the name of a database where a new table will be created
+     *
+     * @param table
+     *   the name of a table to be created
+     *
+     * @param engine
+     *   the name of the MySQL engine for the new table
+     *
+     * @pram partitionByColumn
+     *   (optional, if not empty) the name of a column which will be used
+     *   as a key to configure MySQL partitions for the new table.
+     *   This variation of table schema will be used for the super-transaction-based
+     *   ingest into the table.
+     *
+     * @param columns
+     *   column definitions as pairs of (name,type) of the table
+     *
+     * @param allWorkers
+     *   engage all known workers regardless of their status. If the flag
+     *   is set to 'false' then only 'ENABLED' workers which are not in
+     *   the 'READ-ONLY' state will be involved into the operation.
```

Comment:
No need to mention READ-ONLY.

Reply:
As I explained it earlier, there are two main states for workers `DISABLED` and `ENABLED`, and there are two substates for state `ENABLED`, which are `READ-ONLY` and `READ-WRITE`. In this context I need to be more specific which combination I have in mind here when saying `allWorkers`. This is: `allWorkers := ENABLED & (READ-ONLY or READ-WRITE)`. Otherwise readers might think of `ENABLED or DISABLED`. So, I just wanted to avoid this ambiguity.